lib/device/fan.js

```diff
@@ -87,6 +98,73 @@ export default class {
     }
   }
 
+  async internalSwingUpdate(value) {
```

Comment:
On line 95 above, the characteristic will need to be changed from `On` to `Active` and also the value to `this.cacheState === 'ON' ? 1 : 0`